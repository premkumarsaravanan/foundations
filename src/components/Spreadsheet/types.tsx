import * as React from 'react'
import ReactDataSheet from 'react-datasheet'

export class MyReactDataSheet extends ReactDataSheet<Cell, string> {}

/** Cell contain predefined value
 * https://github.com/nadbm/react-datasheet/blob/master/types/react-datasheet.d.ts
 * plus some properties below
 */
export interface Cell extends ReactDataSheet.Cell<Cell, string> {
  /** The value of the cell, always a string */
  value: string
  /** The validate function, receive Cell as param, must return boolean */
  validate?: (cell: Cell) => boolean
  /** Additional className for styling cell */
  className?: string
  style?: React.CSSProperties
  CustomComponent?: React.FC<{
    data: Cell[][]
    cellRenderProps: ReactDataSheet.CellRendererProps<Cell>
    setData: React.Dispatch<Cell[][]>
    setSelected: React.Dispatch<SelectedMatrix>
  }>
}

export interface CellWithValidate extends Cell {
  validate: (cell: Cell) => boolean
}

export interface DoubleClickPayLoad {
  row: number
  col: number
  maxRowIndex: number
  maxColIndex: number
  isReadOnly?: boolean
}

export interface SpreadsheetProps {
  data: Cell[][]
  description?: React.ReactNode
  hasUploadButton?: boolean
  hasDownloadButton?: boolean
  hasAddButton?: boolean
  /** This will run before data is set to table, must return data with validate function */
  validateUpload?: (data: Cell[][]) => CellWithValidate[][]
}

export interface SelectedMatrix {
  start: ReactDataSheet.Location
  end: ReactDataSheet.Location
}
