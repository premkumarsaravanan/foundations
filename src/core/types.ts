export interface LoginSession {
  accessToken: string
  accessTokenExpiry: number
  idToken: string
  idTokenExpiry: number
  refreshToken: string
}

export interface LoginParams {
  userName: string
  password: string
}

export interface ChangePasswordParams {
  newPassword: string
  userName: string
  password: string
}

export interface RefreshParams {
  userName: string
  refreshToken: string
}

export interface ConfirmPasswordParams {
  userName: string
  newPassword: string
  verificationCode: string
}

export interface ConfirmRegistrationParams {
  userName: string
  verificationCode: string
}

export interface ResetPasswordParams {
  userName: string
}
