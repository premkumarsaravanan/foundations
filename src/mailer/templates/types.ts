export interface ForgottenPasswordTemplateParams {
  verificationCode: string
  userName: string
}

export interface ConfirmPasswordTemplateParams {
  userName: string
}
