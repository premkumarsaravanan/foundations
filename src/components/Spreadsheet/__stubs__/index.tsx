import * as React from 'react'
import { Cell, SelectedMatrix } from '../types'
import ReactDataSheet from 'react-datasheet'

export const data: Cell[][] = [
  [
    { readOnly: true, value: 'Office Name' },
    { readOnly: true, value: 'Building Name' },
    { readOnly: true, value: 'Building No.' },
    { readOnly: true, value: 'Address 1' },
    { readOnly: true, value: 'Address 2' },
    { readOnly: true, value: 'Address 3' },
    { readOnly: true, value: 'Address 4' },
    { readOnly: true, value: 'Post Code' },
    { readOnly: true, value: 'Telephone' },
    { readOnly: true, value: 'Fax' },
    { readOnly: true, value: 'Email' }
  ],
  [
    { value: 'London' },
    { value: 'The White House' },
    { value: '15' },
    { value: 'London 1' },
    { value: '' },
    { value: 'Londom 3' },
    { value: '' },
    { value: 'EC12NH' },
    { value: '0845 0000' },
    { value: '' },
    { value: 'row1@gmail.com' }
  ],
  [
    { value: 'London2' },
    { value: 'The Black House' },
    { value: '11' },
    { value: '' },
    { value: '' },
    { value: 'Adress 3' },
    { value: '' },
    { value: 'EC12NH' },
    { value: '087 471 929' },
    { value: '' },
    { value: 'row2@gmail.com' }
  ],
  [
    { value: 'New York' },
    { value: 'Building A' },
    { value: '11' },
    { value: '' },
    { value: '' },
    { value: 'City Z' },
    { value: '' },
    { value: 'AL7187' },
    { value: '017 7162 9121' },
    { value: '' },
    { value: 'row3@gmail.com' }
  ]
]

export const cellRenderProps: ReactDataSheet.CellRendererProps<Cell> = {
  row: 3,
  col: 10,
  cell: { value: 'row3@gmail.com' },
  selected: false,
  editing: false,
  updated: false,
  attributesRenderer: jest.fn() as ReactDataSheet.AttributesRenderer<Cell>,
  className: 'cell',
  style: { background: 'red' },
  onMouseDown: jest.fn(),
  onMouseOver: jest.fn(),
  onDoubleClick: jest.fn(),
  onContextMenu: jest.fn(),
  children: <div>hi</div>
}

export const setData: React.Dispatch<Cell[][]> = jest.fn()

export const setSelected: React.Dispatch<SelectedMatrix> = jest.fn()

export const selectedMatrix = {
  start: { i: 0, j: 1 },
  end: { i: 2, j: 3 }
}
