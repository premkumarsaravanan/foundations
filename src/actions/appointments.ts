import { actionCreator } from '../utils/actions'
import ActionTypes from '../constants/action-types'
import { AppointmentsData, AppointmentsTime } from '../reducers/appointments'

export interface AppointmentRequestParams {
  time: AppointmentsTime
}

export const appointmentsRequestData = actionCreator<AppointmentRequestParams>(ActionTypes.APPOINTMENTS_REQUEST_DATA)
export const appointmentsRequestDataFailure = actionCreator<void>(ActionTypes.APPOINTMENTS_REQUEST_FAILURE)
export const appointmentsLoading = actionCreator<boolean>(ActionTypes.APPOINTMENTS_LOADING)
export const appointmentsReceiveData = actionCreator<AppointmentsData | undefined>(
  ActionTypes.APPOINTMENTS_RECEIVE_DATA
)
export const appointmentsClearData = actionCreator<null>(ActionTypes.APPOINTMENTS_CLEAR_DATA)
