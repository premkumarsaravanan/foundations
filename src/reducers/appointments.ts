import { Action } from '../types/core'
import { isType } from '../utils/actions'
import {
  appointmentsRequestData,
  appointmentsLoading,
  appointmentsReceiveData,
  appointmentsClearData,
  appointmentsRequestDataFailure
} from '../actions/appointments'
import { PagedResultAppointmentModel_ } from '@/types/appointments'

export type AppointmentsTime = 'Today' | 'Tomorrow' | 'Next week'

export interface AppointmentsData {
  data: PagedResultAppointmentModel_
}

export interface AppointmentsState {
  loading: boolean
  appointments: AppointmentsData | null
  time: AppointmentsTime
}

export const defaultState: AppointmentsState = {
  loading: false,
  appointments: null,
  time: 'Today'
}

const appointmentsReducer = (state: AppointmentsState = defaultState, action: Action<any>): AppointmentsState => {
  if (isType(action, appointmentsRequestData)) {
    return {
      ...state,
      time: action.data.time
    }
  }
  if (isType(action, appointmentsLoading)) {
    return {
      ...state,
      loading: action.data
    }
  }

  if (isType(action, appointmentsReceiveData)) {
    return {
      ...state,
      loading: false,
      appointments: action.data || null
    }
  }

  if (isType(action, appointmentsClearData)) {
    return {
      ...state,
      loading: false,
      appointments: null
    }
  }

  if (isType(action, appointmentsRequestDataFailure)) {
    return {
      ...state,
      loading: false
    }
  }

  return state
}

export default appointmentsReducer
