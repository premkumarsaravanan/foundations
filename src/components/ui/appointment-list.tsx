import React, { memo } from "react";
import { AppointmentTile, getTime, closestTo } from "@reapit/elements";
import { AppointmentModel } from "@/types/appointments";
import CurrentLocButton from "@/components/container/current-loc-button";
import { oc } from "ts-optchain";
import ViewDetailButton from "../container/view-detail-button";
import ETAButton from "./eta-button";
import { NextAppointment } from "@/reducers/next-appointment";

export interface AppointmentListProps {
  data: AppointmentModel[];
  nextAppointment?: NextAppointment | null;
}

export const AppointmentList = memo(
  ({ data, nextAppointment }: AppointmentListProps) => {
    const appointmentNearest = React.useMemo(() => {
      // Get all start date
      const startDatesArray = data.map(item =>
        oc<AppointmentModel>(item).start("")
      );
      // Find appointment nearest current time
      return data.find(
        item => item.start === closestTo(new Date(), startDatesArray)
      );
    }, [data]);

    const refAppointment = React.useRef<HTMLDivElement>(null);

    const [selected, setSelected] = React.useState<AppointmentModel>();

    const onSelectAppointment = (e, item) => {
      if (e.target.type !== "button") {
        if (selected && selected.id === item.id) {
          setSelected(undefined);
        } else {
          setSelected(item);
        }
      }
    };

    React.useEffect(() => {
      if (refAppointment.current) {
        refAppointment.current.scrollIntoView({
          block: "center",
          inline: "center"
        });
      }
    }, []);

    if (data.length === 0) {
      return <div className="py-8 px-3 text-center">No appointments</div>;
    }

    return (
      <div className="px-4">
        {data.map(item => {
          const line1 = oc<AppointmentModel>(item).property.line1("---");
          const line2 = oc<AppointmentModel>(item).property.line2("---");
          const line3 = oc<AppointmentModel>(item).property.line3("---");
          const line4 = oc<AppointmentModel>(item).property.line4("---");
          const postcode = oc<AppointmentModel>(item).property.postcode("---");
          const buildingName = oc<AppointmentModel>(item).property.buildingName(
            "---"
          );
          const buildingNumber = oc<AppointmentModel>(
            item
          ).property.buildingNumber("---");
          const type = oc<AppointmentModel>(item).type("---");
          const start = getTime(oc<AppointmentModel>(item).start(""));
          const end = getTime(oc<AppointmentModel>(item).end(""));

          const hightlight = selected
            ? selected.id === item.id
            : appointmentNearest && appointmentNearest.id === item.id;

          const displayETAButton =
            nextAppointment &&
            nextAppointment.id === item.id &&
            nextAppointment.attendeeHaveMobile &&
            // less than 10 minutes
            nextAppointment.durationValue <= 600;

          let renderETAButton: React.ReactNode = null;

          if (displayETAButton) {
            const tel = oc(nextAppointment)
              .attendeeHaveMobile.communicationDetails([])
              .filter(({ label }) => label === "Mobile")[0].detail;
            const name = oc(nextAppointment).attendeeHaveMobile[0].name("");

            renderETAButton = (
              <div className="mt-4 ml-4">
                <ETAButton
                  tel={tel || ""}
                  body={`Hi ${name}, I will be with you in 10 mins`}
                >
                  ETA Text
                </ETAButton>
              </div>
            );
          }
          return (
            <div
              className="mb-4"
              onClick={e => onSelectAppointment(e, item)}
              key={item.id}
              ref={hightlight ? refAppointment : null}
            >
              <AppointmentTile
                hightlight={hightlight}
                key={item.id}
                heading={line1}
              >
                <ul>
                  <li>Property Address</li>
                  <li>Building Name: {buildingName}</li>
                  <li>Building Number: {buildingNumber}</li>
                  <li>Address 1: {line1}</li>
                  <li>Address 2: {line2}</li>
                  <li>Address 3: {line3}</li>
                  <li>Address 4: {line4}</li>
                  <li>Postcode: {postcode}</li>
                  {/* <li>Contact Name: {name}</li> */}
                  <li>Start Time: {start}</li>
                  <li>End Time: {end}</li>
                  <li>Type of Appointment: {type}</li>
                </ul>
                <div className="flex">
                  <div className="mt-4 mr-4">
                    <ViewDetailButton id={item.id} />
                  </div>
                  <div className="mt-4">
                    <CurrentLocButton />
                  </div>
                  {renderETAButton}
                </div>
              </AppointmentTile>
            </div>
          );
        })}
      </div>
    );
  }
);
