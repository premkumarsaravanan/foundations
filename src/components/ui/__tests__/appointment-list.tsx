import * as React from 'react'
import { AppointmentList } from '../appointment-list'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { appointmentsDataStub } from '@/sagas/__stubs__/appointments'
import { AppointmentModel } from '@/types/appointments'

describe('AppointmentList', () => {
  it('Should match snapshot if having no data', () => {
    expect(toJson(shallow(<AppointmentList data={[]} />))).toMatchSnapshot()
  })

  it('Should match snapshot if having no data', () => {
    expect(
      toJson(shallow(<AppointmentList data={appointmentsDataStub.data.data as AppointmentModel[]} />))
    ).toMatchSnapshot()
  })
})
