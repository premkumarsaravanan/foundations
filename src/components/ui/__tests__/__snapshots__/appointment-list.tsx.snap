// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentList Should match snapshot if having no data 1`] = `
<div
  className="py-8 px-3 text-center"
>
  No appointments
</div>
`;

exports[`AppointmentList Should match snapshot if having no data 2`] = `
<div
  className="px-3"
>
  <AppointmentTile
    heading="High Street"
    key="NEP1600290"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
      </li>
      <li>
        Building Number: 
        56
      </li>
      <li>
        Address 1: 
        High Street
      </li>
      <li>
        Address 2: 
        The Stables
      </li>
      <li>
        Address 3: 
        Old Haversham
      </li>
      <li>
        Address 4: 
        Milton Keynes
      </li>
      <li>
        Postcode: 
        MK19 7DZ
      </li>
      <li>
        Start Time: 
        05:30 PM
      </li>
      <li>
        End Time: 
        06:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="The Embankment"
    key="BED1600537"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        Cornerways
      </li>
      <li>
        Building Number: 
        68
      </li>
      <li>
        Address 1: 
        The Embankment
      </li>
      <li>
        Address 2: 
        Bedford
      </li>
      <li>
        Address 3: 
        Bedfordshire
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        MK40 3PD
      </li>
      <li>
        Start Time: 
        01:30 PM
      </li>
      <li>
        End Time: 
        02:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Sandhill Road"
    key="BUC1600211"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        The Pines
      </li>
      <li>
        Building Number: 
        34
      </li>
      <li>
        Address 1: 
        Sandhill Road
      </li>
      <li>
        Address 2: 
        East Claydon
      </li>
      <li>
        Address 3: 
        Buckingham
      </li>
      <li>
        Address 4: 
        Buckinghamshire
      </li>
      <li>
        Postcode: 
        MK18 2LZ
      </li>
      <li>
        Start Time: 
        02:15 PM
      </li>
      <li>
        End Time: 
        02:45 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Little Tingewick"
    key="BUC1600210"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        Wayside
      </li>
      <li>
        Building Number: 
      </li>
      <li>
        Address 1: 
        Little Tingewick
      </li>
      <li>
        Address 2: 
        Buckingham
      </li>
      <li>
        Address 3: 
        Buckinghamshire
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        MK18 4AG
      </li>
      <li>
        Start Time: 
        01:30 PM
      </li>
      <li>
        End Time: 
        02:15 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="---"
    key="STS1600403"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        ---
      </li>
      <li>
        Building Number: 
        ---
      </li>
      <li>
        Address 1: 
        ---
      </li>
      <li>
        Address 2: 
        ---
      </li>
      <li>
        Address 3: 
        ---
      </li>
      <li>
        Address 4: 
        ---
      </li>
      <li>
        Postcode: 
        ---
      </li>
      <li>
        Start Time: 
        11:15 AM
      </li>
      <li>
        End Time: 
        11:45 AM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="---"
    key="STS1600401"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        ---
      </li>
      <li>
        Building Number: 
        ---
      </li>
      <li>
        Address 1: 
        ---
      </li>
      <li>
        Address 2: 
        ---
      </li>
      <li>
        Address 3: 
        ---
      </li>
      <li>
        Address 4: 
        ---
      </li>
      <li>
        Postcode: 
        ---
      </li>
      <li>
        Start Time: 
        03:15 PM
      </li>
      <li>
        End Time: 
        04:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="High Street"
    key="NEP1600288"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
      </li>
      <li>
        Building Number: 
        56
      </li>
      <li>
        Address 1: 
        High Street
      </li>
      <li>
        Address 2: 
        The Stables
      </li>
      <li>
        Address 3: 
        Old Haversham
      </li>
      <li>
        Address 4: 
        Milton Keynes
      </li>
      <li>
        Postcode: 
        MK19 7DZ
      </li>
      <li>
        Start Time: 
        04:30 PM
      </li>
      <li>
        End Time: 
        05:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="---"
    key="STS1600396"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        ---
      </li>
      <li>
        Building Number: 
        ---
      </li>
      <li>
        Address 1: 
        ---
      </li>
      <li>
        Address 2: 
        ---
      </li>
      <li>
        Address 3: 
        ---
      </li>
      <li>
        Address 4: 
        ---
      </li>
      <li>
        Postcode: 
        ---
      </li>
      <li>
        Start Time: 
        06:00 PM
      </li>
      <li>
        End Time: 
        06:30 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="---"
    key="NEP1600287"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        ---
      </li>
      <li>
        Building Number: 
        ---
      </li>
      <li>
        Address 1: 
        ---
      </li>
      <li>
        Address 2: 
        ---
      </li>
      <li>
        Address 3: 
        ---
      </li>
      <li>
        Address 4: 
        ---
      </li>
      <li>
        Postcode: 
        ---
      </li>
      <li>
        Start Time: 
        05:30 PM
      </li>
      <li>
        End Time: 
        06:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Shelton Road"
    key="BED1600534"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        The Cottage
      </li>
      <li>
        Building Number: 
      </li>
      <li>
        Address 1: 
        Shelton Road
      </li>
      <li>
        Address 2: 
        Yielden
      </li>
      <li>
        Address 3: 
        Bedfordshire
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        MK44 1AQ
      </li>
      <li>
        Start Time: 
        03:15 PM
      </li>
      <li>
        End Time: 
        04:15 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="---"
    key="STS1600394"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        ---
      </li>
      <li>
        Building Number: 
        ---
      </li>
      <li>
        Address 1: 
        ---
      </li>
      <li>
        Address 2: 
        ---
      </li>
      <li>
        Address 3: 
        ---
      </li>
      <li>
        Address 4: 
        ---
      </li>
      <li>
        Postcode: 
        ---
      </li>
      <li>
        Start Time: 
        02:00 PM
      </li>
      <li>
        End Time: 
        02:30 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="The Causeway"
    key="BED1600532"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
      </li>
      <li>
        Building Number: 
        49
      </li>
      <li>
        Address 1: 
        The Causeway
      </li>
      <li>
        Address 2: 
        Carlton
      </li>
      <li>
        Address 3: 
        Bedford
      </li>
      <li>
        Address 4: 
        Bedfordshire
      </li>
      <li>
        Postcode: 
        MK43 7LU
      </li>
      <li>
        Start Time: 
        03:30 PM
      </li>
      <li>
        End Time: 
        04:30 PM
      </li>
      <li>
        Type of Appointment: 
        VL
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Foxfield"
    key="MKC1600312"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
      </li>
      <li>
        Building Number: 
        39
      </li>
      <li>
        Address 1: 
        Foxfield
      </li>
      <li>
        Address 2: 
        Broughton Gate
      </li>
      <li>
        Address 3: 
        Milton Keynes
      </li>
      <li>
        Address 4: 
        Buckinghamshire
      </li>
      <li>
        Postcode: 
        MK10 7BP
      </li>
      <li>
        Start Time: 
        04:30 PM
      </li>
      <li>
        End Time: 
        05:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Bedford Road"
    key="BED1600530"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        The Firs
      </li>
      <li>
        Building Number: 
      </li>
      <li>
        Address 1: 
        Bedford Road
      </li>
      <li>
        Address 2: 
        Northill
      </li>
      <li>
        Address 3: 
        Bedfordshire
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        SG18 9AW
      </li>
      <li>
        Start Time: 
        03:00 PM
      </li>
      <li>
        End Time: 
        05:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Little Tingewick"
    key="BUC1600209"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        Wayside
      </li>
      <li>
        Building Number: 
      </li>
      <li>
        Address 1: 
        Little Tingewick
      </li>
      <li>
        Address 2: 
        Buckingham
      </li>
      <li>
        Address 3: 
        Buckinghamshire
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        MK18 4AG
      </li>
      <li>
        Start Time: 
        01:00 PM
      </li>
      <li>
        End Time: 
        01:30 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="The Embankment"
    key="BED1600525"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        The Homestead
      </li>
      <li>
        Building Number: 
        57
      </li>
      <li>
        Address 1: 
        The Embankment
      </li>
      <li>
        Address 2: 
        Bedford
      </li>
      <li>
        Address 3: 
        Bedfordshire
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        MK40 3PD
      </li>
      <li>
        Start Time: 
        01:30 PM
      </li>
      <li>
        End Time: 
        02:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Tyne Crescent"
    key="BED1600523"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
      </li>
      <li>
        Building Number: 
        54
      </li>
      <li>
        Address 1: 
        Tyne Crescent
      </li>
      <li>
        Address 2: 
        Bedford
      </li>
      <li>
        Address 3: 
        Bedfordshire
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        MK41 7UJ
      </li>
      <li>
        Start Time: 
        11:00 AM
      </li>
      <li>
        End Time: 
        12:00 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="Deacon Place"
    key="MKC1600311"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
      </li>
      <li>
        Building Number: 
        8
      </li>
      <li>
        Address 1: 
        Deacon Place
      </li>
      <li>
        Address 2: 
        Middleton
      </li>
      <li>
        Address 3: 
        Milton Keynes
      </li>
      <li>
        Address 4: 
      </li>
      <li>
        Postcode: 
        MK10 9FS
      </li>
      <li>
        Start Time: 
        11:15 AM
      </li>
      <li>
        End Time: 
        11:30 AM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
  <AppointmentTile
    heading="---"
    key="STS1600389"
  >
    <ul>
      <li>
        Property Address
      </li>
      <li>
        Building Name: 
        ---
      </li>
      <li>
        Building Number: 
        ---
      </li>
      <li>
        Address 1: 
        ---
      </li>
      <li>
        Address 2: 
        ---
      </li>
      <li>
        Address 3: 
        ---
      </li>
      <li>
        Address 4: 
        ---
      </li>
      <li>
        Postcode: 
        ---
      </li>
      <li>
        Start Time: 
        02:00 PM
      </li>
      <li>
        End Time: 
        02:30 PM
      </li>
      <li>
        Type of Appointment: 
        VW
      </li>
    </ul>
    <div
      className="mt-4"
    >
      <CurrentLocButtonWithConnect />
    </div>
  </AppointmentTile>
</div>
`;
